### PR TITLE
Add Method Delete Zone 

### PR DESCRIPTION
**Pull request by benedict-erwin**
Reference link: https://api.cloudflare.com/#zone-delete-zone

### CODE REVIEW

**Comment by benedict-erwin** [comment]
I still cannot resolve this error.

**Comment by beltofte** [comment]
@benedict-erwin Can you trigger a retest, so I can check why it fails?

File: src/Endpoints/Zones.php [hunk 1]
```diff
@@ -233,4 +233,18 @@ public function cachePurge(string $zoneID, array $files = null, array $tags = nu
...
+    public function deleteZone(string $identifier): bool
+    {
+        $user = $this->adapter->delete('zones/' . $identifier);
+        $this->body = json_decode($user->getBody());
+        if (isset($this->body->result->id)) {
+            return true;
+        }
+
+        return false;
+    }
```
Comment (beltofte) [inline src/Endpoints/Zones.php]: This line ends with a tab character.

File: tests/Endpoints/ZoneDeleteTest.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        $mock->expects($this->once())
+            ->method('delete')
+            ->with(
+                $this->equalTo('zones/023e105f4ecef8ad9ca31a8372d0c353')
+            );
+        $zones = new \Cloudflare\API\Endpoints\Zones($mock);
+        $result = $zones->deleteZone('023e105f4ecef8ad9ca31a8372d0c353');
+        $this->assertTrue($result);
+        $this->assertEquals('9a7806061c88ada191ed06f989cc3dac', $zones->getBody()->result->id);
+    }
```
Comment (beltofte) [inline tests/Endpoints/ZoneDeleteTest.php]: This line ends with a trailing space.

File: tests/Endpoints/ZonesTest.php [hunk 3]
```diff
@@ -262,4 +262,5 @@ public function testCachePurge()
         $this->assertTrue($result);
         $this->assertEquals('023e105f4ecef8ad9ca31a8372d0c353', $zones->getBody()->result->id);
     }
+
```
Comment (beltofte) [inline tests/Endpoints/ZonesTest.php]: Remove this new line.

**Comment by beltofte** [comment]
Found the 3 issues in mentioned above when executing "make lint" locally. All of them can be fixed by executing "make fix" as described in CONTRIBUTING.md. After "make fix" is "make lint" passing in my local environment.

**Comment by benedict-erwin** [comment]
> Found the 3 issues in mentioned above when executing "make lint" locally. All of them can be fixed by executing "make fix" as described in CONTRIBUTING.md. After "make fix" is "make lint" passing in my local environment.

Thanks @beltofte

**Comment by IcyApril** [comment]
Thanks for working on this!

**Pull request merged**